### PR TITLE
deepsource: init at 0.8.6

### PR DESCRIPTION
**Pull request by nipeharefa**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/de/deepsource/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchFromGitHub {
+    owner = "DeepSourceCorp";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-6uNb4cQVerrlW/eUkjmlO1i1YKYX3qaVdo0i5cczt+I=";
+  };
+
+  doCheck = false;
+  doInstallCheck = true;
```
Comment (FliegendeWurst) [inline pkgs/by-name/de/deepsource/package.nix:15]:
```suggestion
```

File: pkgs/by-name/de/deepsource/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  vendorHash = "sha256-SsMq4ngq3sSOL28ysHTxTF4CT9sIcCIW7yIhBxIPrNs=";
+
+  ldflags = [ "-s" "-w" "-X=main.version=${version}" ];
+
+  meta = with lib; {
+    description = "Command line interface to DeepSource";
+    mainProgram = "deepsource";
+    homepage = "https://github.com/DeepSourceCorp/cli";
+    license = licenses.bsd3;
```
Comment (FliegendeWurst) [inline pkgs/by-name/de/deepsource/package.nix:25]: Feel free to add yourself as maintainer

File: pkgs/by-name/de/deepsource/package.nix (hunk 1 above)
Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: fixed @FliegendeWurst

File: pkgs/by-name/de/deepsource/package.nix (hunk 2 above)
Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix:24]: done mas @FliegendeWurst

File: pkgs/by-name/de/deepsource/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
```
Comment (FliegendeWurst) [inline pkgs/by-name/de/deepsource/package.nix:24]:
```suggestion
    license = licenses.bsd2;
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/de/deepsource/package.nix (hunk 3 above)
Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: done @FliegendeWurst

File: pkgs/by-name/de/deepsource/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.8.6";
+
+  src = fetchFromGitHub {
+    owner = "DeepSourceCorp";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-6uNb4cQVerrlW/eUkjmlO1i1YKYX3qaVdo0i5cczt+I=";
+  };
+
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deepsource/package.nix:18]: Indicate why in a comment on top of or behind this line.

File: pkgs/by-name/de/deepsource/package.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "-X=main.version=${version}"
+  ];
+
+  meta = with lib; {
+    description = "Command line interface to DeepSource";
+    mainProgram = "deepsource";
+    homepage = "https://github.com/DeepSourceCorp/cli";
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ nipeharefa ];
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deepsource/package.nix:34]:
```suggestion
  meta = {
    description = "Command line interface to DeepSource";
    mainProgram = "deepsource";
    homepage = "https://github.com/DeepSourceCorp/cli";
    license = lib.licenses.bsd2;
    maintainers = with lib.maintainers; [ nipeharefa ];
  };
```

File: pkgs/by-name/de/deepsource/package.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  vendorHash = "sha256-SsMq4ngq3sSOL28ysHTxTF4CT9sIcCIW7yIhBxIPrNs=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=main.version=${version}"
+  ];
+
+  meta = with lib; {
+    description = "Command line interface to DeepSource";
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deepsource/package.nix:29]:
```suggestion
    description = "Command line interface to DeepSource, the code health platform";
```

File: pkgs/by-name/de/deepsource/package.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  doCheck = false;
+
+  vendorHash = "sha256-SsMq4ngq3sSOL28ysHTxTF4CT9sIcCIW7yIhBxIPrNs=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=main.version=${version}"
+  ];
+
```
[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/de/deepsource/package.nix:27]: Also would be good to add a https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook if possible.

**Review by Aleksanaa — COMMENTED** [review]
When you have done all changes, squash/split into two commits only:
1. maintainers: add nipeharefa
2. deepsource: init at 0.8.6

**Comment by nipeharefa** [comment]
done @Aleksanaa

File: pkgs/by-name/de/deepsource/package.nix (hunk 6 above)
Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: done @Aleksanaa

File: pkgs/by-name/de/deepsource/package.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix:30]: fixed @Aleksanaa

File: pkgs/by-name/de/deepsource/package.nix (hunk 4 above)
Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: updated

**Comment by Aleksanaa** [comment]
> done @Aleksanaa

Please check https://github.com/NixOS/nixpkgs/pull/324286/commits/62e73984968c8e1882812caed6d2e35357687940

> Also would be good to add a [nixos.org/manual/nixpkgs/unstable#versioncheckhook](https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook) if possible.

This is not added. Is it possible to do so?

**Comment by nipeharefa** [comment]
oh okeey missing it @Aleksanaa

**Comment by Aleksanaa** [comment]
> done. Please recheck @Aleksanaa

Please check commit https://github.com/NixOS/nixpkgs/pull/324286/commits/ac6b12bc1a2890fbf68d9eaccb23b28d730d02cd.

This commit contains changes from another one, and the order is wrong.

File: pkgs/by-name/de/deepsource/package.nix [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  nativeBuildInputs = [ installShellFiles ];
+  src = fetchFromGitHub {
+    owner = "DeepSourceCorp";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-6uNb4cQVerrlW/eUkjmlO1i1YKYX3qaVdo0i5cczt+I=";
+  };
+
+  doCheck = false; # error when running unit test.
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deepsource/package.nix:22]: Some or all of them? Can you see why (e.g. require certain services, require privileges, require network access)?

We can partially skip test with https://nixos.org/manual/nixpkgs/unstable/#ssec-skip-go-tests.

Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: hi @Aleksanaa updated

**Review by Aleksanaa — APPROVED** [review]
https://github.com/NixOS/nixpkgs/pull/324286/commits/e6203a3f5ec06b4a87b26ab8ae2e52f214a6842c

This is still not resolved. Do you need any help?

**Comment by nipeharefa** [comment]
> [e6203a3](https://github.com/NixOS/nixpkgs/commit/e6203a3f5ec06b4a87b26ab8ae2e52f214a6842c)
> 
> This is still not resolved. Do you need any help?
Hi @Aleksanaa , Did you mean that the commits are not in order or that the names in the maintainer list are not in order?

**Comment by Aleksanaa** [comment]
I mean this commit https://github.com/NixOS/nixpkgs/pull/324286/commits/ac356ed3e03407aeaa7644eaba4978be0d78a063 contains change from the other commit, and the order is wrong

**Comment by nipeharefa** [comment]
> I mean this commit [ac356ed](https://github.com/NixOS/nixpkgs/commit/ac356ed3e03407aeaa7644eaba4978be0d78a063) contains change from the other commit, and the order is wrong

Hi @Aleksanaa , I have pulled the latest changes from the master branch and ran the command 
```
nix-instantiate --eval maintainers/scripts/check-maintainers-sorted.nix
"all good!"
```
is there anything else I need to adjust?

**Comment by Aleksanaa** [comment]
Click on "Commits" on this page, and click on "maintainers: add nipeharefa", see what it contains. Also you have to add yourself to maintainers first, and then add the package, to avoid eval broken on one commit.

**Comment by nipeharefa** [comment]
> Click on "Commits" on this page, and click on "maintainers: add nipeharefa", see what it contains. Also you have to add yourself to maintainers first, and then add the package, to avoid eval broken on one commit.

Okay, could you please check again? The commits have been adjusted.

File: pkgs/by-name/de/deepsource/package.nix [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+}:
+
+buildGoModule rec {
+  pname = "deepsource";
+  version = "0.8.6";
+
+  nativeBuildInputs = [ installShellFiles ];
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deepsource/package.nix:17]: This is not wrong but weird. Could you put this after `postInstall`?

Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix:17]: okey, on it @Aleksanaa

Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: Hi, @Aleksanaa fixed, could you please check again?

File: pkgs/by-name/de/deepsource/package.nix [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  buildGoModule,
+  fetchFromGitHub,
+  versionCheckHook,
+}:
+
+buildGoModule rec {
+  pname = "deepsource";
+  version = "0.8.6";
+
+  nativeBuildInputs = [ installShellFiles ];
```
Comment (Aleksanaa) [inline pkgs/by-name/de/deepsource/package.nix:14]: Sorry, this after `src`

Comment (nipeharefa) [reply pkgs/by-name/de/deepsource/package.nix]: hi, @Aleksanaa fixed

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**